src/features2d/descriptor_matcher.rs

```diff
@@ -64,75 +49,84 @@ impl DescriptorMatcherType {
 
 /// Type for matching keypoint descriptors
 #[derive(Debug)]
-pub struct DescriptorMatcher {
-    value: *mut CDescriptorMatcher,
+pub struct DescriptorMatcher<'a> {
+    value: *mut native::cvsys_DescriptorMatcher,
+    phantom: PhantomData<&'a ()>,
```

Comment:
I'm not sure it's a good idea. We are producing unbound lifetime in the `new` method so it's quite useless.